### PR TITLE
feat: f64 operations

### PR DESCRIPTION
**Pull request by nerodesu017**
### Pull Request Overview

This pull request adds f64-related op codes.

### Testing Strategy

This pull request was tested locally.

### TODO or Help Wanted

N/A

### Formatting

- [x] Ran `cargo fmt`
- [x] Ran `cargo check`
- [x] Ran `cargo build`
- [x] Ran `cargo doc`
- [ ] Ran `nix fmt`
- [ ] Ran `treefmt`

### Github Issue

This pull request implements issue #2

### CODE REVIEW

File: src/execution/mod.rs [hunk 1]
```diff
@@ -24,6 +24,7 @@ pub(crate) mod locals;
 pub(crate) mod store;
 pub mod value;
 pub mod value_stack;
+// pub mod ;
```
Comment (valexandru) [inline src/execution/mod.rs:27]:
```suggestion
```

File: src/core/reader/types/values.rs [hunk 2]
```diff
@@ -40,6 +40,22 @@ impl WasmReader<'_> {
...
+            self.full_wasm_binary[self.pc..self.pc + 8]
+                .try_into()
+                .unwrap(),
+        );
+
+        self.strip_bytes::<8>()?;
+
+        Ok(word)
+    }
+
```
Comment (florianhartung) [inline src/core/reader/types/values.rs:58]: Would this be possible here instead?

```suggestion
        let bytes = self.strip_bytes::<8>().map_err(|_| Error::Eof)?;
        let word = u64::from_le_bytes(bytes);
```

**Review by florianhartung — CHANGES_REQUESTED** [review]

File: src/core/reader/types/values.rs (hunk 2 above)
Comment (florianhartung) [reply src/core/reader/types/values.rs:58]: This should also be possible:
```rs
let bytes = self.strip_bytes::<{ std::mem::size_of::<u64>() }>().map_err(|_| Error::Eof)?;
```

Comment (nerodesu017) [reply src/core/reader/types/values.rs]: Used first suggestion

**Review by florianhartung — APPROVED** [review]

**Review by george-cosma — APPROVED** [review]

**Review by valexandru — APPROVED** [review]

**Pull request merged**